.github/workflows/ci-workflow.yml

```diff
@@ -3,59 +3,33 @@ name: CI Checks
 on: [push, pull_request]
 
 jobs:
-  build:
-    name: Checkout, install, lint, prettier, build and test
+  lint:
+    name: Check audit, lint, and prettier
     runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        node: [ '12', '14', '16' ]
-
     steps:
-      - name: Checkout repository
-        uses: actions/checkout@v2
-
-      - name: Setup Node
-        uses: actions/setup-node@v2
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
         with:
-          node-version: ${{ matrix.node }}
-
-      - name: Get yarn cache directory path
-        id: yarn-cache-dir-path
-        run: echo "::set-output name=dir::$(yarn cache dir)"
-
-      - name: Cache node modules
-        uses: actions/cache@v2
-        id: yarn-cache
-        with:
-          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
-          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
-          restore-keys: |
-            ${{ runner.os }}-yarn-
-
-      - name: Install node dependencies
-        run: yarn install
-
-      - name: Check yarn audit
-        run: yarn audit
-        env:
-          CI: true
-
-      - name: Lint the source code
-        run: yarn lint
+          node-version: '18'
+      - run: npm install
```

Comment:
This probably doesn't matter too much, but I think it is good to use `npm ci` in actions because it uses the versions specified in the `package-lock.json` and I think it's a little faster. Since you're updating the actions now anyway, I figured I'd suggest it now.

If you agree, there's another `npm install` in the `test` action.

Reply:
It's a good idea, but apparently `npm ci` is more sensitive to the `package-lock.json` format -- because it causes the Node 14 actions to fail. Since we still want to target Node 14 (for now), I think we need to stick with `npm install`.

Reply:
Agreed!